tests/utils/mod.rs

```diff
@@ -101,27 +102,84 @@ impl<'a> TestNodeBuilder {
     }
 }
 
+pub const TIMEOUT: Duration = Duration::from_secs(5);
+
 /// Expect that the next event raised by the node matches the given pattern.
 /// Errors if no event, or an event that does not match the pattern is raised.
 #[macro_export]
 macro_rules! expect_next_event {
     ($node:expr, $pattern:pat) => {
-        match tokio::time::timeout(std::time::Duration::from_secs(10), $node.next()).await {
-            Ok(Some($pattern)) => Ok(()),
-            Ok(other) => Err(anyhow::format_err!(
-                "Expecting {}, got {:?}",
-                stringify!($pattern),
-                other
-            )),
-            Err(_) => Err(anyhow::format_err!(
-                "Timeout when expecting {}",
-                stringify!($pattern)
-            )),
+        match tokio::time::timeout($crate::utils::TIMEOUT, $node.next()).await {
+            Ok(Some($pattern)) => {}
+            Ok(other) => panic!("Expecting {}, got {:?}", stringify!($pattern), other),
```

Comment:
Not sure if this was pre/post our chat about returning Result from tests, we should not panic but return an error, is there anything now not allowing you to return a Result?

Reply:
This is maybe up to a philosophical debate, but the way I see it, this macro is essentially a form of assert and asserts do panic. Maybe we can rename it to `assert_next_event` to make it cleaner?

Reply:
ah right i see, I'd vote for renaming it if that's enough for the philosophical debate :smile: , I was seeing it simply as a helper to avoid code duplication